components/feature/addons/src/main/java/mozilla/components/feature/addons/update/AddonUpdater.kt

```diff
@@ -78,6 +79,18 @@ interface AddonUpdater {
         onPermissionsGranted: ((Boolean) -> Unit)
     )
 
+    /**
+     * Register all installed add-ons for future updates, if an add-on is already registered
+     * nothing will happen.
+     */
+    fun registerAllInstalledAddonsForFutureUpdates() {
+        WebExtensionSupport.installedExtensions.values.forEach { extension ->
```

Comment:
See comment below. We should need to read from the map again, and could just pass in the list as a parameter.